src/components/CanvasWrapper.tsx

```diff
@@ -133,17 +133,20 @@ export default function CanvasWrapper(inputProps: CanvasWrapperProps): ReactElem
   const setBackdropVisible = useViewerStateStore((state) => state.setBackdropVisible);
   const setOpenTab = useViewerStateStore((state) => state.setOpenTab);
   const setTrack = useViewerStateStore((state) => state.setTrack);
-  const showHeaderDuringExport = useViewerStateStore((state) => state.showHeaderDuringExport);
-  const showLegendDuringExport = useViewerStateStore((state) => state.showLegendDuringExport);
-  const showScaleBar = useViewerStateStore((state) => state.showScaleBar);
-  const showTimestamp = useViewerStateStore((state) => state.showTimestamp);
+  // const showHeaderDuringExport = useViewerStateStore((state) => state.showHeaderDuringExport);
+  // const showLegendDuringExport = useViewerStateStore((state) => state.showLegendDuringExport);
+  // const showScaleBar = useViewerStateStore((state) => state.showScaleBar);
+  // const showTimestamp = useViewerStateStore((state) => state.showTimestamp);
   const frameLoadResult = useViewerStateStore((state) => state.frameLoadResult);
 
   const containerRef = useRef<HTMLDivElement>(null);
 
   const canv = props.canv;
   const canvasPlaceholderRef = useRef<HTMLDivElement>(null);
 
+  const isFrameLoading = pendingFrame !== currentFrame;
+  const loadProgress = props.loading ? props.loadingProgress : null;
```

Comment:
Updated the loading spinner so it'll now show while frames are loading-- before, it would only show the dataset's load progress. This is useful when loading volumes, which may take longer than normal frames.

Reply:
does this cause flicker for regular 2d datasets?

Reply:
It shouldn't, I think. The loading spinner has a startup delay before it's visible by about a second, so it'll only appear when a frame takes longer than that to load. It also transitions in and out smoothly.